docs/DeveloperGuide.md

```diff
@@ -8,23 +8,128 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Implementation
+
+### Record and RecordList  
+
+The main class in our program is the ```Record``` and ```RecordList``` abstract classes, in which ```Income```, ```Expense``` will inherit from ```Record``` and ```IncomeList``` and ```ExpenseList``` will inherit from ```RecordList```. Most commands will act on instances of the ```Income```, ```Expense```,```IncomeList``` and ```ExpenseList``` classes.  
+
```

Comment:
The notation for abstract class for Record and RecordList is wrong. It should be {Abstract} rather than << Abstract >>. 

Overall, a good detailed class diagram provided